src/base/utils/net.h

```diff
@@ -47,8 +50,12 @@ namespace Utils::Net
     std::optional<Subnet> parseSubnet(const QString &subnetStr);
     bool isIPInSubnets(const QHostAddress &addr, const QList<Subnet> &subnets);
     QString subnetToString(const Subnet &subnet);
+    std::pair<QHostAddress, QHostAddress> subnetToIpRange(const Subnet &subnet);
```

Comment:
I would avoid using generic structs like `pair` whenever possible. We can define `struct Orange { QHostAddress start; QHostAddress end; };`.